tests/certs_test.go

```diff
@@ -286,12 +287,57 @@ var _ = Describe("Certs", func() {
 		})
 
 		Context("multiple domains and certs", func() {
-
 			domains := map[string]string{
 				"wildcard": "*.foo.com",
 				"foo":      "foo.com",
 				"bar":      "bar.com",
 			}
+			domainNames := []string{domains["wildcard"], domains["foo"], domains["bar"]}
+
+			AfterEach(func() {
+				// need to cleanup domains as they are not named randomly as above
+				cleanUpDomains(domainNames)
+			})
+
+			It("can attach/detach 2 certs (1 wildcard) to/from 3 domains (1 wildcard)", func() {
+				sharedCert := certs["wildcard"]
+				certNames = []string{sharedCert.Name, certs["bar"].Name}
+
+				// Add all 3 domains
+				for _, domain := range domains {
+					addDomain(domain, testApp.Name)
+				}
+
+				// Add 2 certs
+				addCert(sharedCert.Name, sharedCert.CertPath, sharedCert.KeyPath)
+				addCert(certs["bar"].Name, certs["bar"].CertPath, certs["bar"].KeyPath)
+
+				// Share wildcard cert betwtixt two domains, attach the other
+				for _, domain := range []string{domains["wildcard"], domains["foo"]} {
+					attachCert(sharedCert.Name, domain)
+				}
+				attachCert(certs["bar"].Name, domains["bar"])
+
+				// With multiple strings to check, use substrings as ordering is non-deterministic
+				// (Should(Say()) enforces strict ordering)
+				bothDomains := fmt.Sprintf("%s,%s", domains["wildcard"], domains["foo"])
+				Eventually(certsInfo(sharedCert.Name).Wait().Out.Contents()).Should(ContainSubstring(bothDomains))
```

Comment:
I don't know why github has weird/non-indentation here.  I've daaablechecked the whitespacing and it looks correct.